compiler/rustc_target/src/spec/mod.rs

```diff
@@ -596,6 +596,18 @@ impl LinkOutputKind {
             _ => return None,
         })
     }
+
+    pub fn can_link_dylib(self) -> bool {
+        match self {
+            LinkOutputKind::StaticNoPicExe
+            | LinkOutputKind::StaticPicExe
+            | LinkOutputKind::StaticDylib
+            | LinkOutputKind::WasiReactorExe => false,
+            LinkOutputKind::DynamicNoPicExe
+            | LinkOutputKind::DynamicPicExe
+            | LinkOutputKind::DynamicDylib => true,
+        }
```

Comment:
```suggestion
        matches!(self, LinkOutputKind::DynamicNoPicExe
            | LinkOutputKind::DynamicPicExe
            | LinkOutputKind::DynamicDylib)
```

Reply:
I prefer explicitly writing out each case, that way this doesn't get missed when a new `LinkOutputKind` variant is added.

Reply:
yes, this is one of those cases where you would want a build failure for new Kind's that have to be handled, so being explicit is better.